runtime/compiler/compile/J9Compilation.cpp

```diff
@@ -651,8 +655,12 @@ J9::Compilation::canAllocateInline(TR::Node* node, TR_OpaqueClassBlock* &classIn
       if (clazz == NULL)
          return -1;
 
+#if defined(JITSERVER_SUPPORT)
       auto classOffset = self()->fej9()->getArrayClassFromComponentClass(self()->fej9()->convertClassPtrToClassOffset(clazz));
       clazz = TR::Compiler->cls.convertClassOffsetToClassPtr(classOffset);
+#else
```

Comment:
The two implementations are identical in functionality. The JITServer solution avoids casts and uses fej9() instead of fej9vm(). I would keep the JITServer one which can be delivered to master separately.
BTW there is some asymmetry between
`fej9()->convertClassPtrToClassOffset(clazz)`
and
`TR::Compiler->cls.convertClassOffsetToClassPtr(classOffset)`
We should use the same API, either `TR::Compiler` or `fej9()`.

Reply:
Will use JITServer implementation. 

> We should use the same API, either TR::Compiler or fej9().

Do you mean adding a new API `J9::ClassEnv::convertClassPtrToClassOffset()` to convert `J9Class` into `TR_OpaqueClassBlock`?

Reply:
Either that or use fej9->convertClassOffsetToClassPointer() which should already exist.

Reply:
`convertClassOffsetToClassPtr()` seems to only exists in `ClassEnv` not in `TR_J9VMBase`. 

Will add a new API `J9::ClassEnv::convertClassPtrToClassOffset()` to convert `J9Class` into `TR_OpaqueClassBlock`.

Reply:
I see this:
```
J9Method *
TR_J9VMBase::convertMethodOffsetToMethodPtr(TR_OpaqueMethodBlock *methodOffset)
   {
   return J9JitMemory::convertMethodOffsetToMethodPtr(methodOffset);
   }

TR_OpaqueMethodBlock *
TR_J9VMBase::convertMethodPtrToMethodOffset(J9Method *methodPtr)
   {
   return J9JitMemory::convertMethodPtrToMethodOffset(methodPtr);
   }
```

Reply:
oops, this is different. It talks about methods not classes